src/Analyzer.TemplateProcessor/ArmTemplateProcessor.cs

```diff
@@ -131,7 +131,7 @@ internal Template ParseAndValidateTemplate(InsensitiveDictionary<JToken> paramet
                 // Do not throw if there was an issue with evaluating language expressions
             }
 
-            template.Resources = ReorderResourceCopies(template, copyNameMap);
+            template.Resources = MapResources(template, copyNameMap);
```

Comment:
Since flatted (sub-) resources also have to be mapped later (line 164), should this method be named `MapTopLevelResources`?